controllers/suite_test.go

```diff
@@ -77,3 +119,224 @@ var _ = AfterSuite(func() {
 	err := testEnv.Stop()
 	Expect(err).ToNot(HaveOccurred())
 })
+
+type MockSpireService struct {
+	entriesById        map[string]*common.RegistrationEntry
+}
+
+func NewMockSpireService() MockSpireService {
+	return MockSpireService{
+		entriesById:        map[string]*common.RegistrationEntry{},
+	}
+}
+
+func (m MockSpireService) getMatchingEntry(in *common.RegistrationEntry) (*common.RegistrationEntry, error) {
+nextEntry:
+	for _, entry := range m.entriesById {
+		if entry.SpiffeId != in.SpiffeId {
+			continue nextEntry
+		}
+
+		if entry.ParentId != in.ParentId {
+			continue nextEntry
+		}
+		if len(entry.Selectors) != len(in.Selectors) {
+			continue nextEntry
+		}
+		var selectorMap map[string]map[string]bool
+		for _, selector := range entry.Selectors {
```

Comment:
Note: if duplicates are allowed in Selectors, funny cases can be constructed maybe with actually different maps, but where this check would pass.

Reply:
Spire rejects duplicate selectors, will add that behaviour to the mock.